src/test/java/software/amazon/cloudformation/resource/IdentifierUtilsTest.java

```diff
@@ -48,4 +48,48 @@ public void generateResourceIdentifier_withShortLength_prefixTruncated() {
         // string that is left to fix the max length.
         assertThat(result).startsWith("my-re-");
     }
+
+    @Test
+    public void generateResourceIdentifier_withStackInvalidStackId() {
+        String result = IdentifierUtils.generateResourceIdentifier("my-stack-name", "my-resource", "123456", 18);
+        assertThat(result.length()).isLessThanOrEqualTo(18);
+
+        // to ensure randomness in the identity, the result will always be a random
+        // string PREFIXED by the size of
+        // string that is left to fix the max length.
+        assertThat(result).startsWith("my-m-");
+    }
+
+    @Test
+    public void generateResourceIdentifier_withStackInvalidStackIdPartitionMismatch() {
+        String result = IdentifierUtils.generateResourceIdentifier("stack/my-stack-name", "my-resource", "123456", 50);
```

Comment:
I don't think we should support it. CloudFormation stack name never have `/`.  If customer want to use their own prefix, they should not use this method. 

let's make it as fail to throw error if the stackId like this.

Reply:
but this will never happen I guess, if stack arn is submitted then pattern will match otherwise linter will reject malformed stack names